qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -0,0 +1,274 @@
+        backend_properties = pass_manager_config.backend_properties
+        routing_pass = BasicSwap(coupling_map)
+        vf2_call_limit = common.get_vf2_call_limit(
+            opt_level, pass_manager_config.layout_method, pass_manager_config.initial_layout
+        )
+        if opt_level == 0:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map=coupling_map,
+                seed_transpiler=seed_transpiler,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 1:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map,
+                vf2_call_limit=vf2_call_limit,
+                backend_properties=backend_properties,
+                seed_transpiler=seed_transpiler,
+                check_trivial=True,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 2:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map=coupling_map,
+                vf2_call_limit=vf2_call_limit,
+                backend_properties=backend_properties,
+                seed_transpiler=seed_transpiler,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 3:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map=coupling_map,
+                vf2_call_limit=vf2_call_limit,
+                backend_properties=backend_properties,
+                seed_transpiler=seed_transpiler,
+                use_barrier_before_measurement=True,
+            )
+        else:
+            raise TranspilerError(f"Invalid optimization level specified: {opt_level}")
+
+
+class StochasticSwapPassManager(PassManagerStagePlugin):
+    """Plugin class for routing stage with :class:`~.StochasticSwap`"""
+
+    def pass_manager(self, pass_manager_config: PassManagerConfig) -> PassManager:
+        """Build routing stage PassManager."""
+        opt_level = pass_manager_config.optimization_level
+        seed_transpiler = pass_manager_config.seed_transpiler
+        target = pass_manager_config.target
+        coupling_map = pass_manager_config.coupling_map
+        backend_properties = pass_manager_config.backend_properties
+        vf2_call_limit = common.get_vf2_call_limit(
+            opt_level, pass_manager_config.layout_method, pass_manager_config.initial_layout
+        )
+        if opt_level == 3:
+            routing_pass = StochasticSwap(coupling_map, trials=200, seed=seed_transpiler)
+        else:
+            routing_pass = StochasticSwap(coupling_map, trials=20, seed=seed_transpiler)
+
+        if opt_level == 0:
+            return common.generate_routing_passmanager(
```

Comment:
I think that's probably a reasonable addition to those common functions, although I'd probably opt to make all the arguments optional so a pass manager config isn't required (as users find creating one by hand annoying if they're just trying to get the PassManager by hand) and error if insufficient input is provided. But I think we can work on evolving that interface independently of this PR. Especially since those functions are already part of the API so making changes to that can happen in parallel.